src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenForManagedIdentityParameterBuilder.cs

```diff
@@ -59,6 +59,25 @@ public AcquireTokenForManagedIdentityParameterBuilder WithForceRefresh(bool forc
             return this;
         }
 
+        /// <summary>
+        /// Sets claims in the query. Use when the AAD admin has enabled conditional access evaluation. Acquiring the token normally will result in a
+        /// <see cref="MsalUiRequiredException"/> with the <see cref="MsalServiceException.Claims"/> property set. Retry the 
+        /// token acquisition, and use this value in the <see cref="WithClaims(string)"/> method. See https://aka.ms/msal-exceptions for details
+        /// as well as https://aka.ms/msal-net-claim-challenge.
+        /// </summary>
+        /// <param name="claims">A string with one or multiple claims.</param>
+        /// <returns>The builder to chain .With methods.</returns>
+        public AcquireTokenForManagedIdentityParameterBuilder WithClaims(string claims)
+        {
+            if (string.IsNullOrEmpty(claims))
+            {
+                throw new ArgumentNullException(nameof(claims));
+            }
+
+            Parameters.Claims = claims;
```

Comment:
You forgot to update `ManagedIdentityAuthRequest` (the one that derives from `RequestBase`) to take into account `Claims`. So it does not bypass teh token cache.